### DISCUSSION TITLE
fast.farm queastion

### DISCUSSION THREAD

**Discussion by clun1** [OpenFAST/openfast#2126] [General]
If there is any problem with the simulation output, the following is the position of the wind turbine and the output result. The inlet wind speed of the wind turbine after the first wind turbine is close, which seems to be the result of inconsistent with the wake
![image](https://github.com/OpenFAST/openfast/assets/93849049/a079a7c2-c7f0-409a-b9de-f65a781280fe)
![image](https://github.com/OpenFAST/openfast/assets/93849049/e6dc70c0-785f-4487-9e59-91691c1b40a1)
![image](https://github.com/OpenFAST/openfast/assets/93849049/8b21fafe-74f3-4b6f-8017-87a12e40bfab)
![image](https://github.com/OpenFAST/openfast/assets/93849049/006e4dd5-0fd4-49e4-8d69-8b2e96af034e)
![image](https://github.com/OpenFAST/openfast/assets/93849049/64e29a7e-6bda-4c5a-8253-5806b1181460)
![image](https://github.com/OpenFAST/openfast/assets/93849049/0a674a6f-4401-410c-bee1-4fecd7b3db6e)
![image](https://github.com/OpenFAST/openfast/assets/93849049/ffbe9039-6db8-4687-9332-08e2338feb6a)
![image](https://github.com/OpenFAST/openfast/assets/93849049/63af59f1-8a5a-4bd8-89c6-9f5ea7f55b92)

**Comment by jjonkman** [comment]
Dear @clun1,

I'm not sure I understand your question.

FYI: It looks like your wind input has some built-in periodicity.

Best regards,

**Reply by clun1** [reply]
Dear Jonkman

My doubt is that the downstream wind turbine should have less output power than the upstream wind turbine, because the further the wind turbine will be subject to the wake superposition of the upstream wind turbine, thus reducing the inflow wind speed of the downstream wind turbine, thus reducing the output power of the downstream wind turbine.But the picture shows a different outcome.

Best regards,

**Reply by jjonkman** [reply]
Dear @clun1,

If I understand correctly, your concern is that the power is not continually dropping down the row of turbines?  This is a bit hard to tell from the graph you shared (it may help to compute/compare the time-series mean of power).  Regardless, your result is not too surprising either, because I'd expect the wake effect to settile out along the row eventually.  The upstream turbine is clearly extracting the most power and has the strongest wake, which impacts that second turbine that extracts less power but has a weaker wake, etc.

Best regards,

**Reply by clun1** [reply]
Dear Jonkman,

Thank you for your answer. Thank you for your answer. There is another question. I used openfast_toolbox to generate wind farm parameters, the following are the alarms. In addition, the operation time of wind farms is greatly extended, and it takes 12 hours to run 5 wind turbines for 1000 seconds. Is that normal?
<img width="450" alt="image" src="https://github.com/OpenFAST/openfast/assets/93849049/ebcef6d8-f832-4f0b-a4f9-25f17bfe7223">

Best regards,

**Reply by jjonkman** [reply]
Dear @clun1,

This warning from ServoDyn will be displayed when using the Bladed-style DISCON controller interface in OpenFAST if your glue code time step (`DT`) is not the same as your controller time step (`DLL_DT`), or if your OpenFAST model is invoking correction steps (`NumCrctn` > 0)

You mention that the wind farm simulation time is "greatly extended", but you haven't said what your comparing to and what you changed.  The computational time will also depend on if you running FAST.Farm in serial mode or parallel mode, as well as the spatial-temporal discretization of the model.

Best regards,
